### DISCUSSION TITLE
Registration results in downsampling?

### DISCUSSION THREAD

**Discussion by archang** [CoBrALab/RABIES#273] [General]
Hello,

I initially thought that commonspace registration referred to resampling to the reference atlas' dimensions (keeping EPI voxel resolution unchanged, as explained in the docs). However, when I look at the dimensions of my cleaned data, the resolution has been downsampled, rather than upsampled to the reference atlas' dimensions. For instance, my input data might have dimensions of 96x96x20x300, and my reference atlas dimensions of 256x256x65. Yet, the commonspace registration has dimensions of 70x70x21x300. This downsampling results in some of the smaller regions in my reference atlas being lost.

For reference, I am using RABIES with the --bold_only option.

**Comment by gdevenyi** [comment]
Please provide the full command used, and the details of any custom files provided during invocation.

**Reply by archang** [reply]
Thank you for the assistance.

Here is what I ran for preprocess:
All of the anat_template, masks, & label inputs have dimensions of 256x256x65.

```
230131-21:18:30,473 nipype.workflow INFO:
	 Running RABIES - version: 0.4.8
230131-21:18:30,473 nipype.workflow INFO:
	 CLI INPUTS: 
-> rabies_stage = preprocess 
-> plugin = MultiProc 
-> local_threads = 8 
-> scale_min_memory = 1.0 
-> min_proc = 1 
-> verbose = 1 
-> bids_dir = /input_BIDS 
-> output_dir = /preprocess_outputs 
-> bold_only = True 
-> anat_autobox = False 
-> bold_autobox = False 
-> apply_despiking = False 
-> HMC_option = intraSubjectBOLD 
-> apply_slice_mc = False 
-> detect_dummy = False 
-> data_type = float32 
-> anat_inho_cor = {'method': 'SyN', 'otsu_thresh': '2', 'multiotsu': False} 
-> anat_robust_inho_cor = {'apply': False, 'masking': False, 'brain_extraction': False, 'template_registration': 'SyN'} 
-> bold_inho_cor = {'method': 'Rigid', 'otsu_thresh': '2', 'multiotsu': False} 
-> bold_robust_inho_cor = {'apply': False, 'masking': False, 'brain_extraction': False, 'template_registration': 'SyN'} 
-> commonspace_reg = {'masking': False, 'brain_extraction': False, 'template_registration': 'SyN', 'fast_commonspace': False} 
-> bold2anat_coreg = {'masking': False, 'brain_extraction': False, 'registration': 'SyN'} 
-> nativespace_resampling = inputs_defined 
-> commonspace_resampling = inputs_defined 
-> anatomical_resampling = inputs_defined 
-> apply_STC = True 
-> TR = 3.0,masking=true,brain_extraction=false,template_registration=SyN,fast_commonspace=false 
-> tpattern = alt-z 
-> stc_axis = Y 
-> anat_template = /home/rabies/rat/Atlas_ref.nii.gz 
-> brain_mask = /home/rabies/rat/Atlas_bin.nii.gz 
-> WM_mask = /home/rabies/rat/rat_wm_mask.nii.gz 
-> CSF_mask = /home/rabies/rat/rat_ventricle_mask.nii.gz 
-> vascular_mask = /home/rabies/rat/Atlas_bin.nii.gz 
-> labels = /home/rabies/rat/Atlas.nii.gz 
```

**Comment by gdevenyi** [comment]
Can you provide the ants `PrintHeader` or an fsl-equivalent of the header info for the input files

**Reply by archang** [reply]
Header info for one of the subjects: [input_sub.txt](https://github.com/CoBrALab/RABIES/files/10758552/input_sub.txt)
Header info for atlas: [Atlas_dim.txt](https://github.com/CoBrALab/RABIES/files/10758553/Atlas_dim.txt)

**Reply by archang** [reply]
Header info for one of the subject output (from bold_datasink/commonspace_bold): 
[output_sub.txt](https://github.com/CoBrALab/RABIES/files/10758626/output_sub.txt)

**Reply by Gab-D-G** [reply]
The resolution of the image was indeed unchanged (voxel resolution is 1x0.31x0.31 in both input and output images); only the FOV was cropped to match the template. If registration was succesful (inspect the preprocess_QC_report), this should have accurately isolated the brain voxels in your EPI image.

**Comment by Gab-D-G** [comment]
Hi, the resampling resolution refers to voxel size in mm, not the number of voxels defining the FOV. From the options you provided, the EPI should have kept the same voxel resolution, but the FOV gets cropped to match the dimensions of the reference template (or the anatomical scans for native space resampling). You should inspect the header of the outputs and see whether the voxel resolution was changed.
